plugin/clang.vim

```diff
@@ -875,9 +935,20 @@ func! s:ClangExecute(root, clang_options, line, col)
   let l:command .= ' 1>'.l:tmps[0].' 2>'.l:tmps[1]
   let l:res = [[], []]
   if has("nvim")
-    let l:argv = ['sh', '-c', l:command]
-    call s:PDebug("s:ClangExecute::job.argv", l:argv, 2)
-    call jobstart(l:argv, {'fstdout': l:tmps[0], 'fstderr': l:tmps[1], 'on_exit': function('ClangExecuteNeoJobHandler')})
+    let b:clang_memtmps = [[],[]]
+
+    let l:nvimsrc = getline(1, '$')
+    call add(l:nvimsrc, "\<cr>")
+
+    let l:nvimcmd = printf('%s -cc1 -x c++ -fsyntax-only -code-completion-macros -code-completion-at -:%d:%d %s',
+                      \ g:clang_exec, a:line, a:col, a:clang_options)
+    let l:argv = l:nvimcmd
+
+    " set env
+    let l:jobid = jobstart(['sh', '-c', l:argv], s:neojobcallbacks)
+    call jobsend(l:jobid, l:nvimsrc)
+    call jobclose(l:jobid, 'stdin')
```

Comment:
Good job! We can send source content to clang in nvim now!
@mlen here's a solution provided by @syswow !